src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java

```diff
@@ -218,6 +196,93 @@
         }
     }
 
+    private ResponseEntity<?> getListedSubscriptions(String subscriptionName) {
+        // set is used to prevent subscription names repeating
+        Set<String> subscriptionNameList = new HashSet<>(Arrays.asList(subscriptionName.split(",")));
+        List<Subscription> foundSubscriptionList = new ArrayList<>();
+        List<String> notFoundSubscriptionList = new ArrayList<>();
+
+        subscriptionNameList.forEach(name -> {
+            try {
+                LOGGER.debug("Subscription fetching has been started: {}", name);
+                // Make sure the password is not sent outside this service.
+                Subscription subscription = subscriptionService.getSubscription(name);
+                subscription.setPassword("");
+                foundSubscriptionList.add(subscription);
+                LOGGER.debug("Subscription [{}] fetched successfully.", name);
+            } catch (SubscriptionNotFoundException e) {
+                LOGGER.error("Subscription not found: {}", name);
+                LOGGER.debug("Subscription not found traceback:\n {}", e);
+                notFoundSubscriptionList.add(name);
+            } catch (Exception e) {
+                LOGGER.error("Failed to fetch subscription {}", name);
+                LOGGER.debug("Failed to fetch subscription:\n {}", e);
+                notFoundSubscriptionList.add(name);
+            }
+        });
+        GetSubscriptionResponse response = new GetSubscriptionResponse();
+        response.setFoundSubscriptions(foundSubscriptionList);
+        response.setNotFoundSubscriptions(notFoundSubscriptionList);
+        HttpStatus httpStatus = (!foundSubscriptionList.isEmpty()) ? HttpStatus.OK : HttpStatus.NOT_FOUND;
```

Comment:
Personally I'm not really fond of these oneliners ... I'd prefer them to be spelled out, making them easier to read :)

Reply:
I made this into an if statement.